test/mitmproxy/test_examples.py

```diff
@@ -28,3 +47,59 @@ def test_load_scripts():
                 raise
         else:
             s.unload()
+
+
+def test_add_header():
+    flow = tutils.tflow(resp=netutils.tresp())
+    add_header.response({}, flow)
+    assert flow.response.headers["newheader"] == "foo"
+
+
+def test_modify_form():
+    form_header = Headers(content_type="application/x-www-form-urlencoded")
+    flow = tutils.tflow(req=netutils.treq(headers=form_header))
+    modify_form.request({}, flow)
+    assert flow.request.urlencoded_form["mitmproxy"] == ["rocks"]
+
+
+def test_modify_querystring():
+    flow = tutils.tflow(req=netutils.treq(path="/search?q=term"))
+    modify_querystring.request({}, flow)
+    assert flow.request.query["mitmproxy"] == ["rocks"]
+
+
+def test_modify_response_body():
+    ctx = DummyContext()
+    tutils.raises(ValueError, modify_response_body.start, ctx, [])
+
+    modify_response_body.start(ctx, ["modify-response-body.py", "mitmproxy", "rocks"])
+    assert ctx.old == "mitmproxy" and ctx.new == "rocks"
+
+    flow = tutils.tflow(resp=netutils.tresp(content="I <3 mitmproxy"))
+    modify_response_body.response(ctx, flow)
+    assert flow.response.content == "I <3 rocks"
+
+
+def test_custom_contentviews():
+    pig = custom_contentviews.ViewPigLatin()
```

Comment:
@mhils: A downside of the `example()` contextmanager approach, I can't test this out! 

And am getting an error, when I try to load this via the contextmanager.

```
>               raise ScriptException(traceback.format_exc(e))
E               ScriptException: Traceback (most recent call last):
E                 File "/home/dufferzafar/dev/alpha/gsoc16/honeynet/mitmproxy/mitmproxy/script/script.py", line 96, in run
E                   return f(self.ctx, *args, **kwargs)
E                 File "/home/dufferzafar/dev/alpha/gsoc16/honeynet/mitmproxy/examples/custom_contentviews.py", line 64, in start
E                   context.add_contentview(pig_view)
E                 File "/home/dufferzafar/dev/alpha/gsoc16/honeynet/mitmproxy/mitmproxy/script/script_context.py", line 57, in add_contentview
E                   contentviews.add(view_obj)
E                 File "/home/dufferzafar/dev/alpha/gsoc16/honeynet/mitmproxy/mitmproxy/contentviews.py", line 521, in add
E                   raise ContentViewException("Duplicate view: " + view.name)
E               ContentViewException: Duplicate view: pig_latin_HTML
```